transport/http/request.go

```diff
@@ -38,6 +38,32 @@ func (r *Request) Clone() *Request {
 	return &rc
 }
 
+// StreamLength returns the number of bytes of the serialized stream attached
+// to the request and ok set. If the length cannot be determined, an error will
+// be returned.
+func (r *Request) StreamLength() (size int64, ok bool, err error) {
+	if l, ok := r.stream.(interface{ Len() int }); ok {
+		return int64(l.Len()), true, nil
+	}
+
+	if !r.isStreamSeekable {
+		return 0, false, nil
+	}
+
+	s := r.stream.(io.Seeker)
+	endOffset, err := s.Seek(0, io.SeekEnd)
+	if err != nil {
+		return 0, false, err
+	}
+
+	_, err = s.Seek(r.streamStartPos, io.SeekStart)
+	if err != nil {
+		return 0, false, err
+	}
```

Comment:
Can you comment an example for when streamStartPos < io.seekStart? 

I thought io.SeekStart will always be less than streamStartPos.

Reply:
`SeekStart` will start from the beginning of the stream and seek `streamStartPos` bytes forward.  The reason to seek to `streamStartPos` instead of `0` is to ensure that the SDK only sends the stream from the starting position the user's application provided it to the SDK at. For example application opens a file, and wants to skip the first N bytes uploading the rest. The application would move the file's offset N bytes, then hand it off to the SDK to send the remaining.